### ISSUE TITLE
Support for nested group claims 

### ISSUE THREAD

**Issue by rysiekpl** [zorn-v/nextcloud-social-login#116]
In certain scenarios it would be good to be able to use nested group claims. For instance, Keycloak returns client roles (which can be used for group access) as:

```
 "resource_access": {
    "client-id": {
      "roles": [
        "client-role-1",
        "client-role-2",
        (...)
      ]
    },
    (...)
}
 ```

It would be good to be able to specify the groups claim as comma-separated nested address, in this case `resource_access.client-id.roles`.

**Comment by zorn-v** [comment]
I think about it

**Comment by zorn-v** [comment]
How keycloak should be configured to return such response ?

**Comment by rysiekpl** [comment]
> How keycloak should be configured to return such response?

Keycloak provides the `resource_access` claim by default, as far as I know. No specific configuration required, apart from having client roles defined for a given client and assigned to a given user.

> Check v2.1.0

Checked, got an error:

```
The server was unable to complete your request.

If this happens again, please send the technical details below to the server administrator.

More details can be found in the server log.
Technical details

    Remote Address: 194.105.229.47
    Request ID: LjOBBZcVH9NNSe3TMeYr
    Type: Error
    Code: 0
    Message: Cannot use object of type stdClass as array
    File: /opt/nextcloud/apps/sociallogin/lib/Provider/CustomOAuth2.php
    Line: 76


Trace

#0 /opt/nextcloud/apps/sociallogin/lib/Provider/CustomOpenIDConnect.php(38): OCA\SocialLogin\Provider\CustomOAuth2->getGroups(Object(Hybridauth\Data\Collection))
#1 /opt/nextcloud/apps/sociallogin/lib/Controller/LoginController.php(276): OCA\SocialLogin\Provider\CustomOpenIDConnect->getUserProfile()
#2 /opt/nextcloud/apps/sociallogin/lib/Controller/LoginController.php(185): OCA\SocialLogin\Controller\LoginController->auth('OCA\\SocialLogin...', Array, 'secure-occrp-or...', 'OpenID Connect')
#3 /opt/nextcloud/lib/private/AppFramework/Http/Dispatcher.php(166): OCA\SocialLogin\Controller\LoginController->customOidc('secure-occrp-or...')
#4 /opt/nextcloud/lib/private/AppFramework/Http/Dispatcher.php(99): OC\AppFramework\Http\Dispatcher->executeController(Object(OCA\SocialLogin\Controller\LoginController), 'customOidc')
#5 /opt/nextcloud/lib/private/AppFramework/App.php(126): OC\AppFramework\Http\Dispatcher->dispatch(Object(OCA\SocialLogin\Controller\LoginController), 'customOidc')
#6 /opt/nextcloud/lib/private/AppFramework/Routing/RouteActionHandler.php(47): OC\AppFramework\App::main('OCA\\SocialLogin...', 'customOidc', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)
#7 [internal function]: OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)
#8 /opt/nextcloud/lib/private/Route/Router.php(297): call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)
#9 /opt/nextcloud/lib/base.php(975): OC\Route\Router->match('/apps/sociallog...')
#10 /opt/nextcloud/index.php(42): OC::handleRequest()
#11 {main}

```

Relevant part of `var_dump($data)` in `CustomOAuth2->getGroups()`:
```
object(Hybridauth\Data\Collection)#2805 (1) {
  ["collection":protected]=> object(stdClass)#2800 (20) {
    "0" ["resource_access"]=> object(stdClass)#2802 (2) {
      ["test-nas-occrp-org"]=> object(stdClass)#2804 (1) {
        ["roles"]=> array(1) {
          [0]=> string(11) "group_test1" }
        }
      }
    }
  } 
```

**Comment by rysiekpl** [comment]
Also: seems like I needed to enable `Add to ID Token` on `Client Roles` mapper in the Keycloak Client config to get this to get the claim.

`Add to Access Token` is enabled by default, and I also tried `Add to Userinfo`. Neither worked.

And thank you for working on this one!

**Comment by zorn-v** [comment]
> Checked, got an error:

Fixed in v2.1.1

**Comment by rysiekpl** [comment]
Yup, I can confirm the fix works. Thank you!
